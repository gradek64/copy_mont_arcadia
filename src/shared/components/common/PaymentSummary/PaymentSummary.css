/** @define PaymentSummary */
.PaymentSummary {
  max-width: $payment-summary-max-width;
  border-top: $payment-summary-border-top-width solid $payment-summary-border-top-color;
  border-right: $payment-summary-border-right-width solid $payment-summary-border-right-color;
  border-bottom: $payment-summary-border-bottom-width solid $payment-summary-border-bottom-color;
  border-left: $payment-summary-border-left-width solid $payment-summary-border-left-color;
  padding-top: $payment-summary-padding-top;
}

.PaymentSummary-header {
  height: 40px;
  border-bottom: 1px solid $payment-summary-header-border-bottom;
}

.PaymentSummary-headerText {
  margin: 0;
  color: $payment-summary-header-text-color;
  font-size: $payment-summary-header-font-size;
  font-weight: $payment-summary-header-font-weight;
  line-height: $payment-summary-header-line-height;
  text-transform: $payment-summary-header-text-transform;
}

.PaymentSummary-tableHead > .PaymentSummary-tableRow > .PaymentSummary-tableColumn {
  font-weight: $payment-summary-table-head-font-weight;
}

.PaymentSummary-body {
  background-color: $payment-summary-body-background-color;
  padding-top: $payment-summary-body-padding-top;
  padding-right: $payment-summary-body-padding-right;
  padding-bottom: $payment-summary-body-padding-bottom;
  padding-left: $payment-summary-body-padding-left;
}

.PaymentSummary-table {
  width: 100%;
  border-collapse: collapse;
}

.PaymentSummary-tableRow {
  padding: 0;
  font-size: $payment-summary-table-row-font-size;
  font-family: $payment-summary-table-row-font-family;
  font-weight: $payment-summary-table-row-font-weight;
  line-height: $payment-summary-table-row-line-height;
  letter-spacing: $payment-summary-table-row-letter-spacing;
  border-bottom: $payment-summary-table-row-border-bottom;
}

.PaymentSummary-tableColumn {
  padding: $payment-summary-table-col-padding;
}

.PaymentSummary-totalRow {
  text-align: left;
  & > .PaymentSummary-tableColumn {
    padding: $payment-summary-total-col-padding;
    border-bottom: $payment-summary-total-border-bottom;
  }
}


.PaymentSummary-tableColumnRight {
  text-align: right;
  white-space: nowrap;
  font-size: $payment-summary-table-col-right-font-size;
  font-weight: $payment-summary-table-col-right-font-weight;
  line-height: $payment-summary-table-col-right-line-height;
}

