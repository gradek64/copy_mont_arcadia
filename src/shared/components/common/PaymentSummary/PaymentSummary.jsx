import React from 'react'
import PropTypes from 'prop-types'
import { PrivacyGuard } from '../../../lib/privacy-guard'
import { lastFourCharsOf } from '../../../lib/string-utils'
import { GCARD } from '../../../../shared/constants/paymentTypes'
import GiftCardsRemainingBalance from './GiftCardsRemainingBalance'

export default function PaymentSummary(props, context) {
  const { totalOrderPrice, payments } = props
  const { l, p } = context
  const paymentHasGiftCards = payments.some(
    (paymentMethod) => paymentMethod.type === GCARD
  )
  return (
    <article className="PaymentSummary">
      <header className="PaymentSummary-header">
        <h1 className="PaymentSummary-headerText qa-heading">{l`Payment details`}</h1>
      </header>
      <div className="PaymentSummary-body">
        <table className="PaymentSummary-table">
          <thead className="PaymentSummary-tableHead qa-thead">
            <tr className="PaymentSummary-tableRow qa-row PaymentSummary-totalRow">
              <th className="PaymentSummary-tableColumn qa-column">{l`Total Cost`}</th>
              <th className="PaymentSummary-tableColumn PaymentSummary-tableColumnRight PaymentSummary-totalPrice qa-column">
                {p(totalOrderPrice)}
              </th>
            </tr>
          </thead>
          <tbody className="qa-tbody">
            {payments.map((payment, i) => (
              <tr
                key={i} // eslint-disable-line react/no-array-index-key
                className="PaymentSummary-tableRow qa-row"
              >
                <td className="PaymentSummary-tableColumn qa-column">
                  <PrivacyGuard>
                    <span>{payment.method}</span>
                  </PrivacyGuard>
                  {payment.isCardType && (
                    <PrivacyGuard>
                      <span>
                        **** **** **** {lastFourCharsOf(payment.cardNumber)}
                      </span>
                    </PrivacyGuard>
                  )}
                </td>
                <td className="PaymentSummary-tableColumn PaymentSummary-tableColumnRight qa-column">
                  <PrivacyGuard>
                    <span>
                      {p(payment.priceAfterDiscount || payment.price)}
                    </span>
                  </PrivacyGuard>
                </td>
              </tr>
            ))}
            {paymentHasGiftCards && (
              <GiftCardsRemainingBalance payments={payments} />
            )}
          </tbody>
        </table>
      </div>
    </article>
  )
}

PaymentSummary.contextTypes = {
  l: PropTypes.func,
  p: PropTypes.func,
}

PaymentSummary.propTypes = {
  payments: PropTypes.arrayOf(
    PropTypes.shape({
      method: PropTypes.string,
      cardNumber: PropTypes.string,
      price: PropTypes.string,
      type: PropTypes.string.isRequired,
    })
  ),
  totalOrderPrice: PropTypes.string,
}
