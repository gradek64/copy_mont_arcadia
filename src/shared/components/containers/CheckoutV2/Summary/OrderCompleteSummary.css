/** @define OrderCompleteSummary */

.OrderCompleteSummary {
  margin: $order-complete-padding auto;
  max-width: $laptop-end-width;

  &-error {
    text-align: $order-complete-error-text-align;

    & .OrderCompleteSummary-icon {
      display: $order-complete-error-icon-display;
      margin-bottom: $order-complete-error-icon-margin-bottom;
    }

    & .OrderCompleteSummary-errorImage {
      display: block;
      margin: 0 auto;
      width: $order-complete-error-icon-width;
    }

    & .OrderCompleteSummary-header {
      display: $order-complete-error-header-display;
      margin-top: 0;
      margin-bottom: $order-complete-error-header-margin-bottom;
      font-size: $order-complete-error-header-font-size;
      font-weight: $order-complete-error-header-font-weight;
    }
  }
  &-payments {
    margin: 25px 0;
  }
  &-row {
    display: flex;
    justify-content: space-between;
    margin: 5px 0;
  }
  &-trackMyOrder {
    margin: 15px 0;
  }
  &-storeButton {
    margin: 15px 0;
  }
  &-trackMyOrderButton {
    width: 100%;
  }
}

@media $min-tablet {
  .OrderCompleteSummary {
    margin: $checkout-container-margin-vertical auto;
    padding-left: $page-padding;
    padding-right: $page-padding;
    &-container {
      display: flex;
      width: 100%;
      flex-wrap: wrap;
    }
    &-left {
      width: 100%;
      flex: 1;
      padding-right: 20px;
    }
    &-header {
      background: $order-complete-header-background-color;
      padding: $order-complete-header-padding;
      text-transform: uppercase;
      width: 100%;
    }
    &-success {
      align-items: center;
      display: flex;
      flex-direction: column;
    }
    &-trackMyOrder {
      margin-top: 0;
    }
    &-trackMyOrderButton {
      width: 100%;
    }
  }
}

@media $min-laptop {
  .OrderCompleteSummary {
    &-bottom {
      max-width: 1200px;
    }
    &-trackMyOrderButton {
      width: auto;
    }
  }
}
