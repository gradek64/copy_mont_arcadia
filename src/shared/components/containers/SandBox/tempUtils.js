// The following have been added in order to permit fallback to CmsWrapper for the rendering of the Components
// that mrCMS is not yet capable to manage.
import { path, flatten } from 'ramda'

// Added a copy of existig function in /server (parseCmsForm, buildValidationSchema, setValidationRules) since we temporarely need it. Didn't move the function to /shared
// since these functions will be removed once mrCMS gets capable to manage TermsAndConditions, CmsForms, Campaigns.
function setValidationRules(fields, field, validationSchema) {
  if (!validationSchema[field.name]) validationSchema[field.name] = []
  Object.keys(field.validation).forEach((validator) => {
    // This maps the CMS forms validation keys to our validator functions.
    // It has to be serialized data for the redux store.

    if (field.validation[validator]) {
      switch (validator) {
        case 'equalTo': {
          const fieldGUID = field.validation[validator].condition
          const fieldNameToEqual = fields.find(({ guid }) => guid === fieldGUID)
            .name
          const fieldErrorMessage = field.validation[validator].message
          validationSchema[field.name] = validationSchema[field.name].concat({
            isSameAs: [fieldErrorMessage, fieldNameToEqual],
          })
          break
        }
        default:
          validationSchema[field.name] = validationSchema[field.name].concat(
            validator
          )
          break
      }
    }
  })
}
function buildValidationSchema(fieldsets) {
  const validationSchema = {}
  fieldsets.forEach(({ fields }) => {
    fields.forEach((field) => {
      if (field.dropdowns) {
        const dropdowns = path(['dropdowns'], field)
        dropdowns.forEach((dropdown) =>
          setValidationRules(fields, dropdown, validationSchema)
        )
      } else if (field.name) {
        const validation = path(['validation'], field)
        if (validation) {
          setValidationRules(fields, field, validationSchema)
        }
      }
    })
  })
  return validationSchema
}
export function parseCmsForm(content) {
  return {
    ...content,
    fieldSchema: flatten(
      content.fieldsets.map((fieldset) => {
        return fieldset.fields
          .map((field) => {
            if (field.dropdowns)
              return field.dropdowns.map((dropdown) => dropdown.name)
            return field.name
          })
          .filter((name) => name)
      })
    ),
    validationSchema: buildValidationSchema(content.fieldsets),
  }
}
