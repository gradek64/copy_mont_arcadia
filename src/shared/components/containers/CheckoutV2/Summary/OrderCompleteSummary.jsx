import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import Helmet from 'react-helmet'
import { contains, isEmpty } from 'ramda'
import { normalizePrice } from '../../../../lib/price'

// Selectors
import {
  extractDiscountInfo,
  getDeliveryDate,
  getSubTotal,
  isCollectFromOrder,
} from '../../../../selectors/checkoutSelectors'
import {
  isDDPStandaloneOrderCompleted,
  getDDPDefaultName,
  isDDPOrderCompleted,
} from '../../../../selectors/ddpSelectors'
import { getPaymentMethods } from '../../../../selectors/paymentMethodSelectors'
import {
  getCurrencyCode,
  getBrandName,
  getLogoVersion,
} from '../../../../selectors/configSelectors'
import { getStoreLocatorStores } from '../../../../selectors/storeLocatorSelectors'
import { isMobile } from '../../../../selectors/viewportSelectors'

// Components
import Espot from '../../Espot/Espot'
import PaymentSummary from '../../../common/PaymentSummary/PaymentSummary'
import OrderCompleteButton from '../../../common/OrderCompleteButton/OrderCompleteButton'
import OrderCompleteBasket from '../../../common/OrderCompleteBasket/OrderCompleteBasket'
import OrderCompleteDelivery from '../../../common/OrderCompleteDelivery/OrderCompleteDelivery'
import SandBox from '../../SandBox/SandBox'
import GuestRegisterForm from '../Guest/GuestRegisterForm'

// Constants
import { GCARD } from '../../../../constants/paymentTypes'
import espotsMobile from '../../../../constants/espotsMobile'
import cmsConsts from '../../../../constants/cmsConsts'

const mapStateToProps = (state) => ({
  brandName: getBrandName(state),
  currencyCode: getCurrencyCode(state),
  deliveryDate: getDeliveryDate(state),
  ddpDefaultName: getDDPDefaultName(state),
  discountInfo: extractDiscountInfo(state),
  isCollectFromOrder: isCollectFromOrder(state),
  isDDPOrderCompleted: isDDPOrderCompleted(state),
  isDDPStandaloneOrderCompleted: isDDPStandaloneOrderCompleted(state),
  isMobile: isMobile(state),
  logoVersion: getLogoVersion(state),
  minLaptop: contains(state.viewport.media, ['laptop', 'desktop']),
  orderSubtotal: getSubTotal(state),
  paymentMethods: getPaymentMethods(state),
  stores: getStoreLocatorStores(state),
})

const OrderCompleteSummary = (
  {
    brandName,
    buttonClickHandler,
    deliveryDate,
    ddpDefaultName,
    discountInfo,
    isCollectFromOrder,
    isDDPOrderCompleted,
    isDDPStandaloneOrderCompleted,
    isDiscoverMoreEnabled,
    isMobile,
    logoVersion,
    minLaptop,
    orderCompleted,
    orderError,
    orderSubtotal,
    paymentMethods,
    stores,
  },
  { l }
) => {
  if (!orderCompleted || isEmpty(orderCompleted)) {
    return null
  }

  const payments = orderCompleted.paymentDetails.map((payment) => {
    // Checking the discount is applied to a payment method other than gift card
    const {
      cardNumberStar,
      selectedPaymentMethod,
      totalCost,
      totalCostAfterDiscount,
      paymentMethod,
      remainingBalance,
    } = payment

    // Extracting the payment method from the paymentConfig
    const method = paymentMethods.find((method) => {
      return (
        selectedPaymentMethod !== GCARD &&
        method.value === selectedPaymentMethod
      )
    })
    const isThirdPartyPaymentMethod = method && method.type === 'OTHER'

    const isDiscounted =
      (cardNumberStar || isThirdPartyPaymentMethod) &&
      totalCostAfterDiscount &&
      normalizePrice(totalCost) !== normalizePrice(totalCostAfterDiscount)

    return {
      method: paymentMethod,
      cardNumber: cardNumberStar,
      price: normalizePrice(totalCost),
      type: selectedPaymentMethod,
      isCardType: !isThirdPartyPaymentMethod,
      // priceAfterDiscount is returned only when a gift card is applied
      ...(isDiscounted && {
        priceAfterDiscount: normalizePrice(totalCostAfterDiscount),
      }),
      ...(selectedPaymentMethod === GCARD && {
        remainingBalance,
      }),
    }
  })

  const renderMobileEspot = () => (
    <SandBox
      cmsPageName={espotsMobile.orderConfirmation[0]}
      contentType={cmsConsts.ESPOT_CONTENT_TYPE}
      isInPageContent
    />
  )

  return (
    <section className="OrderCompleteSummary">
      <Helmet title={l`Your Order is Complete`} />
      {!orderError && (
        <div className="OrderCompleteSummary-success">
          <div className="OrderCompleteSummary-container">
            <div className="OrderCompleteSummary-left">
              <OrderCompleteDelivery
                brandName={brandName}
                ddpDefaultName={ddpDefaultName}
                deliveryDate={deliveryDate}
                orderCompleted={orderCompleted}
                isDDPStandaloneOrderCompleted={isDDPStandaloneOrderCompleted}
                isCollectFromOrder={isCollectFromOrder}
                isDDPOrderCompleted={isDDPOrderCompleted}
                logoVersion={logoVersion}
                minLaptop={minLaptop}
                stores={stores}
              />
              <Espot identifier="thankyouESpot1" />
              {isMobile && renderMobileEspot()}
              {orderCompleted.isGuestOrder &&
                !orderCompleted.isRegisteredEmail && (
                  <GuestRegisterForm l={l} />
                )}
              <PaymentSummary
                payments={payments}
                totalOrderPrice={orderCompleted.totalOrderPrice}
              />
              <Espot identifier="thankyouESpot2" />
              <Espot identifier="thankyouESpot3" />
            </div>
            <OrderCompleteBasket
              discountInfo={discountInfo}
              orderCompleted={orderCompleted}
              orderSubtotal={orderSubtotal}
              isDDPStandaloneOrderCompleted={isDDPStandaloneOrderCompleted}
              isMobile={isMobile}
            />
          </div>
          <OrderCompleteButton
            buttonClickHandler={buttonClickHandler}
            isDiscoverMoreEnabled={isDiscoverMoreEnabled}
            orderError={orderError}
            orderCompleted={orderCompleted}
          />
        </div>
      )}
    </section>
  )
}

OrderCompleteSummary.contextTypes = {
  l: PropTypes.func,
}

OrderCompleteSummary.propTypes = {
  brandName: PropTypes.string.isRequired,
  buttonClickHandler: PropTypes.func.isRequired,
  ddpDefaultName: PropTypes.string.isRequired,
  isDDPStandaloneOrderCompleted: PropTypes.bool.isRequired,
  isDiscoverMoreEnabled: PropTypes.bool.isRequired,
  logoVersion: PropTypes.string.isRequired,
  minLaptop: PropTypes.bool,
  orderCompleted: PropTypes.object.isRequired,
  paymentMethods: PropTypes.array.isRequired,
  deliveryDate: PropTypes.string,
  discountInfo: PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.number,
      label: PropTypes.string,
    })
  ),
  isCollectFromOrder: PropTypes.bool,
  isDDPOrderCompleted: PropTypes.bool,
  isMobile: PropTypes.bool,
  orderError: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  orderSubtotal: PropTypes.string,
  stores: PropTypes.arrayOf(
    PropTypes.shape({
      storeId: PropTypes.string,
      latitude: PropTypes.number,
      longitude: PropTypes.number,
    })
  ),
}

OrderCompleteSummary.defaultProps = {
  currencyCode: 'GBP',
  discountInfo: [],
  deliveryDate: '',
  isCollectFromOrder: false,
  isDDPOrderCompleted: false,
  isMobile: true,
  minLaptop: false,
  orderError: false,
  orderSubtotal: '',
  stores: [],
}

export default connect(mapStateToProps)(OrderCompleteSummary)
