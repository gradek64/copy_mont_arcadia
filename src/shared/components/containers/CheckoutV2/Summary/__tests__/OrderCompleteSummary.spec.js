import testComponentHelper from 'test/unit/helpers/test-component'
import OrderCompleteSummary from '../OrderCompleteSummary'
import PaymentSummary from '../../../../common/PaymentSummary/PaymentSummary'
import SandBox from '../../../SandBox/SandBox'
import espotsMobile from '../../../../../constants/espotsMobile'

const initialProps = {
  buttonClickHandler: jest.fn(),
  isDiscoverMoreEnabled: true,
  orderCompleted: {
    deliveryCost: '4.00',
    deliveryPrice: '4.01',
    orderId: 3464614,
    subTotal: '33.00',
    paymentDetails: [
      {
        paymentMethod: 'Visa',
        cardNumberStar: '************1111',
        totalCost: '£30.00',
        totalCostAfterDiscount: '£10.00',
        selectedPaymentMethod: 'VISA',
      },
    ],
  },
  orderError: false,
  brandName: 'topshop',
  deliveryDate: 'Wednesday 10 June 2020',
  ddpDefaultName: 'Topshop Premier',
  isDDPStandaloneOrderCompleted: false,
  isMobile: false,
  logoVersion: '19102018',
  minLaptop: true,
  orderSubtotal: '33.00',
  paymentMethods: [],
}

const mappedPayments = [
  {
    cardNumber: '************1111',
    isCardType: true,
    method: 'Visa',
    price: '30.00',
    priceAfterDiscount: '10.00',
    type: 'VISA',
  },
]

describe('<OrderCompleteSummary />', () => {
  const renderComponent = testComponentHelper(
    OrderCompleteSummary.WrappedComponent
  )

  describe('@render', () => {
    describe('Order is complete with no errors', () => {
      const component = renderComponent(initialProps)
      const { wrapper } = renderComponent(initialProps)

      it('default', () => {
        expect(component.getTree()).toMatchSnapshot()
      })

      it('should pass down the correct `payments` prop', () => {
        expect(wrapper.find(PaymentSummary).props().payments).toEqual(
          mappedPayments
        )
      })

      it('should not show mobile espot', () => {
        expect(wrapper.find(SandBox).exists()).toBeFalsy()
      })

      describe('is mobile', () => {
        const component = renderComponent({
          ...initialProps,
          isMobile: true,
        })
        const { wrapper } = component
        it('should show the mobile espot', () => {
          expect(component.getTree()).toMatchSnapshot()
          expect(wrapper.find(SandBox).props().cmsPageName).toEqual(
            espotsMobile.orderConfirmation[0]
          )
        })
      })
    })
    describe('Order has error', () => {
      it('should render title only', () => {
        const component = renderComponent({
          ...initialProps,
          orderError: true,
        })
        expect(component.getTree()).toMatchSnapshot()
      })
    })
    describe('Order is not complete', () => {
      it('should render nothing', () => {
        const { wrapper } = renderComponent({
          ...initialProps,
          orderCompleted: {},
        })
        expect(wrapper.html()).toBeNull()
      })
    })
    describe('Guest Checkout', () => {
      it('should render the registration form if it is a guest order with a non-registered email', () => {
        const component = renderComponent({
          ...initialProps,
          orderCompleted: {
            ...initialProps.orderCompleted,
            isGuestOrder: true,
            isRegisteredEmail: false,
          },
        })
        expect(component.getTree()).toMatchSnapshot()
      })
      it('should not render the registration form if it is a guest order with a registered email', () => {
        const component = renderComponent({
          ...initialProps,
          orderCompleted: {
            ...initialProps.orderCompleted,
            isGuestOrder: true,
            isRegisteredEmail: true,
          },
        })
        expect(component.getTree()).toMatchSnapshot()
      })
    })
  })
})
