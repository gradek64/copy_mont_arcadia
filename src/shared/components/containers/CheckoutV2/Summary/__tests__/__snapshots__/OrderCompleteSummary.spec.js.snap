// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderCompleteSummary /> @render Guest Checkout should not render the registration form if it is a guest order with a registered email 1`] = `
<section
  className="OrderCompleteSummary"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Your Order is Complete"
  />
  <div
    className="OrderCompleteSummary-success"
  >
    <div
      className="OrderCompleteSummary-container"
    >
      <div
        className="OrderCompleteSummary-left"
      >
        <OrderCompleteDelivery
          brandName="topshop"
          ddpDefaultName="Topshop Premier"
          deliveryDate="Wednesday 10 June 2020"
          isCollectFromOrder={false}
          isDDPOrderCompleted={false}
          isDDPStandaloneOrderCompleted={false}
          logoVersion="19102018"
          minLaptop={true}
          orderCompleted={
            Object {
              "deliveryCost": "4.00",
              "deliveryPrice": "4.01",
              "isGuestOrder": true,
              "isRegisteredEmail": true,
              "orderId": 3464614,
              "paymentDetails": Array [
                Object {
                  "cardNumberStar": "************1111",
                  "paymentMethod": "Visa",
                  "selectedPaymentMethod": "VISA",
                  "totalCost": "£30.00",
                  "totalCostAfterDiscount": "£10.00",
                },
              ],
              "subTotal": "33.00",
            }
          }
          stores={Array []}
        />
        <Connect(Espot)
          identifier="thankyouESpot1"
        />
        <PaymentSummary
          payments={
            Array [
              Object {
                "cardNumber": "************1111",
                "isCardType": true,
                "method": "Visa",
                "price": "30.00",
                "priceAfterDiscount": "10.00",
                "type": "VISA",
              },
            ]
          }
        />
        <Connect(Espot)
          identifier="thankyouESpot2"
        />
        <Connect(Espot)
          identifier="thankyouESpot3"
        />
      </div>
      <OrderCompleteBasket
        discountInfo={Array []}
        isDDPStandaloneOrderCompleted={false}
        isMobile={false}
        orderCompleted={
          Object {
            "deliveryCost": "4.00",
            "deliveryPrice": "4.01",
            "isGuestOrder": true,
            "isRegisteredEmail": true,
            "orderId": 3464614,
            "paymentDetails": Array [
              Object {
                "cardNumberStar": "************1111",
                "paymentMethod": "Visa",
                "selectedPaymentMethod": "VISA",
                "totalCost": "£30.00",
                "totalCostAfterDiscount": "£10.00",
              },
            ],
            "subTotal": "33.00",
          }
        }
        orderSubtotal="33.00"
      />
    </div>
    <OrderCompleteButton
      buttonClickHandler={[MockFunction]}
      isDiscoverMoreEnabled={true}
      orderCompleted={
        Object {
          "deliveryCost": "4.00",
          "deliveryPrice": "4.01",
          "isGuestOrder": true,
          "isRegisteredEmail": true,
          "orderId": 3464614,
          "paymentDetails": Array [
            Object {
              "cardNumberStar": "************1111",
              "paymentMethod": "Visa",
              "selectedPaymentMethod": "VISA",
              "totalCost": "£30.00",
              "totalCostAfterDiscount": "£10.00",
            },
          ],
          "subTotal": "33.00",
        }
      }
      orderError={false}
    />
  </div>
</section>
`;

exports[`<OrderCompleteSummary /> @render Guest Checkout should render the registration form if it is a guest order with a non-registered email 1`] = `
<section
  className="OrderCompleteSummary"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Your Order is Complete"
  />
  <div
    className="OrderCompleteSummary-success"
  >
    <div
      className="OrderCompleteSummary-container"
    >
      <div
        className="OrderCompleteSummary-left"
      >
        <OrderCompleteDelivery
          brandName="topshop"
          ddpDefaultName="Topshop Premier"
          deliveryDate="Wednesday 10 June 2020"
          isCollectFromOrder={false}
          isDDPOrderCompleted={false}
          isDDPStandaloneOrderCompleted={false}
          logoVersion="19102018"
          minLaptop={true}
          orderCompleted={
            Object {
              "deliveryCost": "4.00",
              "deliveryPrice": "4.01",
              "isGuestOrder": true,
              "isRegisteredEmail": false,
              "orderId": 3464614,
              "paymentDetails": Array [
                Object {
                  "cardNumberStar": "************1111",
                  "paymentMethod": "Visa",
                  "selectedPaymentMethod": "VISA",
                  "totalCost": "£30.00",
                  "totalCostAfterDiscount": "£10.00",
                },
              ],
              "subTotal": "33.00",
            }
          }
          stores={Array []}
        />
        <Connect(Espot)
          identifier="thankyouESpot1"
        />
        <GuestRegisterForm
          l={
            [MockFunction] {
              "calls": Array [
                Array [
                  Array [
                    "Your Order is Complete",
                  ],
                ],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": "Your Order is Complete",
                },
              ],
            }
          }
        />
        <PaymentSummary
          payments={
            Array [
              Object {
                "cardNumber": "************1111",
                "isCardType": true,
                "method": "Visa",
                "price": "30.00",
                "priceAfterDiscount": "10.00",
                "type": "VISA",
              },
            ]
          }
        />
        <Connect(Espot)
          identifier="thankyouESpot2"
        />
        <Connect(Espot)
          identifier="thankyouESpot3"
        />
      </div>
      <OrderCompleteBasket
        discountInfo={Array []}
        isDDPStandaloneOrderCompleted={false}
        isMobile={false}
        orderCompleted={
          Object {
            "deliveryCost": "4.00",
            "deliveryPrice": "4.01",
            "isGuestOrder": true,
            "isRegisteredEmail": false,
            "orderId": 3464614,
            "paymentDetails": Array [
              Object {
                "cardNumberStar": "************1111",
                "paymentMethod": "Visa",
                "selectedPaymentMethod": "VISA",
                "totalCost": "£30.00",
                "totalCostAfterDiscount": "£10.00",
              },
            ],
            "subTotal": "33.00",
          }
        }
        orderSubtotal="33.00"
      />
    </div>
    <OrderCompleteButton
      buttonClickHandler={[MockFunction]}
      isDiscoverMoreEnabled={true}
      orderCompleted={
        Object {
          "deliveryCost": "4.00",
          "deliveryPrice": "4.01",
          "isGuestOrder": true,
          "isRegisteredEmail": false,
          "orderId": 3464614,
          "paymentDetails": Array [
            Object {
              "cardNumberStar": "************1111",
              "paymentMethod": "Visa",
              "selectedPaymentMethod": "VISA",
              "totalCost": "£30.00",
              "totalCostAfterDiscount": "£10.00",
            },
          ],
          "subTotal": "33.00",
        }
      }
      orderError={false}
    />
  </div>
</section>
`;

exports[`<OrderCompleteSummary /> @render Order has error should render title only 1`] = `
<section
  className="OrderCompleteSummary"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Your Order is Complete"
  />
</section>
`;

exports[`<OrderCompleteSummary /> @render Order is complete with no errors default 1`] = `
<section
  className="OrderCompleteSummary"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Your Order is Complete"
  />
  <div
    className="OrderCompleteSummary-success"
  >
    <div
      className="OrderCompleteSummary-container"
    >
      <div
        className="OrderCompleteSummary-left"
      >
        <OrderCompleteDelivery
          brandName="topshop"
          ddpDefaultName="Topshop Premier"
          deliveryDate="Wednesday 10 June 2020"
          isCollectFromOrder={false}
          isDDPOrderCompleted={false}
          isDDPStandaloneOrderCompleted={false}
          logoVersion="19102018"
          minLaptop={true}
          orderCompleted={
            Object {
              "deliveryCost": "4.00",
              "deliveryPrice": "4.01",
              "orderId": 3464614,
              "paymentDetails": Array [
                Object {
                  "cardNumberStar": "************1111",
                  "paymentMethod": "Visa",
                  "selectedPaymentMethod": "VISA",
                  "totalCost": "£30.00",
                  "totalCostAfterDiscount": "£10.00",
                },
              ],
              "subTotal": "33.00",
            }
          }
          stores={Array []}
        />
        <Connect(Espot)
          identifier="thankyouESpot1"
        />
        <PaymentSummary
          payments={
            Array [
              Object {
                "cardNumber": "************1111",
                "isCardType": true,
                "method": "Visa",
                "price": "30.00",
                "priceAfterDiscount": "10.00",
                "type": "VISA",
              },
            ]
          }
        />
        <Connect(Espot)
          identifier="thankyouESpot2"
        />
        <Connect(Espot)
          identifier="thankyouESpot3"
        />
      </div>
      <OrderCompleteBasket
        discountInfo={Array []}
        isDDPStandaloneOrderCompleted={false}
        isMobile={false}
        orderCompleted={
          Object {
            "deliveryCost": "4.00",
            "deliveryPrice": "4.01",
            "orderId": 3464614,
            "paymentDetails": Array [
              Object {
                "cardNumberStar": "************1111",
                "paymentMethod": "Visa",
                "selectedPaymentMethod": "VISA",
                "totalCost": "£30.00",
                "totalCostAfterDiscount": "£10.00",
              },
            ],
            "subTotal": "33.00",
          }
        }
        orderSubtotal="33.00"
      />
    </div>
    <OrderCompleteButton
      buttonClickHandler={[MockFunction]}
      isDiscoverMoreEnabled={true}
      orderCompleted={
        Object {
          "deliveryCost": "4.00",
          "deliveryPrice": "4.01",
          "orderId": 3464614,
          "paymentDetails": Array [
            Object {
              "cardNumberStar": "************1111",
              "paymentMethod": "Visa",
              "selectedPaymentMethod": "VISA",
              "totalCost": "£30.00",
              "totalCostAfterDiscount": "£10.00",
            },
          ],
          "subTotal": "33.00",
        }
      }
      orderError={false}
    />
  </div>
</section>
`;

exports[`<OrderCompleteSummary /> @render Order is complete with no errors is mobile should show the mobile espot 1`] = `
<section
  className="OrderCompleteSummary"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Your Order is Complete"
  />
  <div
    className="OrderCompleteSummary-success"
  >
    <div
      className="OrderCompleteSummary-container"
    >
      <div
        className="OrderCompleteSummary-left"
      >
        <OrderCompleteDelivery
          brandName="topshop"
          ddpDefaultName="Topshop Premier"
          deliveryDate="Wednesday 10 June 2020"
          isCollectFromOrder={false}
          isDDPOrderCompleted={false}
          isDDPStandaloneOrderCompleted={false}
          logoVersion="19102018"
          minLaptop={true}
          orderCompleted={
            Object {
              "deliveryCost": "4.00",
              "deliveryPrice": "4.01",
              "orderId": 3464614,
              "paymentDetails": Array [
                Object {
                  "cardNumberStar": "************1111",
                  "paymentMethod": "Visa",
                  "selectedPaymentMethod": "VISA",
                  "totalCost": "£30.00",
                  "totalCostAfterDiscount": "£10.00",
                },
              ],
              "subTotal": "33.00",
            }
          }
          stores={Array []}
        />
        <Connect(Espot)
          identifier="thankyouESpot1"
        />
        <Connect(SandBox)
          cmsPageName="mobileOrderConfirmationESpotPos1"
          contentType="espot"
          isInPageContent={true}
        />
        <PaymentSummary
          payments={
            Array [
              Object {
                "cardNumber": "************1111",
                "isCardType": true,
                "method": "Visa",
                "price": "30.00",
                "priceAfterDiscount": "10.00",
                "type": "VISA",
              },
            ]
          }
        />
        <Connect(Espot)
          identifier="thankyouESpot2"
        />
        <Connect(Espot)
          identifier="thankyouESpot3"
        />
      </div>
      <OrderCompleteBasket
        discountInfo={Array []}
        isDDPStandaloneOrderCompleted={false}
        isMobile={true}
        orderCompleted={
          Object {
            "deliveryCost": "4.00",
            "deliveryPrice": "4.01",
            "orderId": 3464614,
            "paymentDetails": Array [
              Object {
                "cardNumberStar": "************1111",
                "paymentMethod": "Visa",
                "selectedPaymentMethod": "VISA",
                "totalCost": "£30.00",
                "totalCostAfterDiscount": "£10.00",
              },
            ],
            "subTotal": "33.00",
          }
        }
        orderSubtotal="33.00"
      />
    </div>
    <OrderCompleteButton
      buttonClickHandler={[MockFunction]}
      isDiscoverMoreEnabled={true}
      orderCompleted={
        Object {
          "deliveryCost": "4.00",
          "deliveryPrice": "4.01",
          "orderId": 3464614,
          "paymentDetails": Array [
            Object {
              "cardNumberStar": "************1111",
              "paymentMethod": "Visa",
              "selectedPaymentMethod": "VISA",
              "totalCost": "£30.00",
              "totalCostAfterDiscount": "£10.00",
            },
          ],
          "subTotal": "33.00",
        }
      }
      orderError={false}
    />
  </div>
</section>
`;
